client.go

```diff
@@ -448,8 +446,8 @@ func (c *Client) connect(identifier, serverAddr string) error {
 	c.log.Debug("client has started successfully.")
 	c.redialBackoff.Reset() // we successfully connected, so we can reset the backoff
 
-	c.mu.Lock()
-	if c.startNotify != nil && !c.closed {
+	c.closedMu.RLock()
+	if !c.closed && c.startNotify != nil {
```

Comment:
hm, sounds interesting, but then this operation should be Lock not RLock and we need to change the documentation on top of the "closedMu" and the parameter name itself, since it will be used for another purpose. 

What do you guys think @mmatczuk @rjeczalik ?